src/main/java/it/gov/pagopa/pu/workflow/wf/exportfile/wfexportfile/ExportFileWFImpl.java

```diff
@@ -0,0 +1,111 @@
+import it.gov.pagopa.pu.processexecutions.dto.generated.ExportFileStatus;
+import it.gov.pagopa.pu.workflow.wf.exportfile.activity.ScheduleExportFileExpirationActivity;
+import it.gov.pagopa.pu.workflow.wf.exportfile.config.ExportFileWFConfig;
+import java.time.LocalDate;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.BeansException;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.context.ApplicationContext;
+import org.springframework.context.ApplicationContextAware;
+
+@Slf4j
+@WorkflowImpl(taskQueues = ExportFileWFImpl.TASK_QUEUE_EXPORT_FILE_WF)
+public class ExportFileWFImpl implements ExportFileWF, ApplicationContextAware {
+  public static final String TASK_QUEUE_EXPORT_FILE_WF = "ExportFileWF";
+  public static final String TASK_QUEUE_EXPORT_FILE_LOCAL_ACTIVITY = "ExportFileWF_LOCAL";
+
+  @Value("${schedule.export-file-expiration.days}")
+  private int expirationDays;
+
+  private ExportFileActivity exportFileActivity;
+  private UpdateExportFileStatusActivity updateExportFileStatusActivity;
+  private ScheduleExportFileExpirationActivity scheduleExportFileExpirationActivity;
+
+  /**
+   * Temporal workflow will not allow to use injection in order to avoid <a href="https://docs.temporal.io/workflows#non-deterministic-change">non-deterministic changes</a> due to dynamic reconfiguration.<BR />
+   * Anyway it allows to override ActivityOptions, but actually it's not supporting the override based on the particular workflow.<BR />
+   * In {@link it.gov.pagopa.pu.workflow.config.TemporalWFImplementationCustomizer} we are already setting defaults to all workflows.<BR />
+   * Use this as an example to override based on the particular workflow.
+   */
+  @Override
+  public void setApplicationContext(ApplicationContext applicationContext) throws BeansException {
+    ExportFileWFConfig wfConfig = applicationContext.getBean(ExportFileWFConfig.class);
+    exportFileActivity = wfConfig.buildExportFileActivityStub();
+    updateExportFileStatusActivity = wfConfig.buildUpdateExportFileStatusActivityStub();
+    scheduleExportFileExpirationActivity = wfConfig.buildScheduleExportFileExpirationActivityStub();
+  }
+
+  @Override
+  public void exportFile(Long exportFileId, ExportFileTypeEnum exportFileType) {
+    log.info("Starting export file workflow for exportFileId: {}", exportFileId);
+    updateExportFileStatus(UpdateStatusRequest.builder()
+      .exportFileId(exportFileId)
+      .oldStatus(ExportFileStatus.REQUESTED)
+      .newStatus(ExportFileStatus.PROCESSING)
+      .build());
+
+    ExportFileResult exportFileResult = null;
+    String errorDescription = null;
+    try {
+      exportFileResult = exportFileActivity.executeExport(
+        exportFileId, exportFileType);
+    } catch(Exception e){
+      errorDescription = e.getMessage();
+    }
+
+    LocalDate expirationDate = LocalDate.now().plusDays(expirationDays);
```

Comment:
to get the LocalDate.now() it should be better to have it stored on Temporal in order to avoid a value change due to a retry on a different date of the same WF.

We could use Side Effects (https://docs.temporal.io/develop/java/side-effects)
```suggestion
    LocalDate expirationDate = Workflow.sideEffect(LocalDate.class, () -> LocalDate.now()).plusDays(expirationDays);
```

or we could add the exportDate on the ExportFileResult DTO